js-packages/@prestojs/viewmodel/src/ViewModelCache.ts

```diff
@@ -488,6 +885,33 @@ export default class ViewModelCache<
         this.cache = new Map();
     }
 
+    /**
+     * Batch changes made within provided function. This guarantees that any changes made
+     * will result in a single call for each relevant listener.
+     *
+     * ```js
+     * User.cache.addListener(listenerAll);
+     * User.cache.addListener(1, ['id', 'name'], listener);
+     * User.cache.addListenerList([1, 2], ['id', 'name'], listenerList);
+     * User.cache.batch(() => {
+     *   // This value won't appear in changes at all as it's replaced 2 lines down
+     *   User.cache.add({ id: 1, name: 'Bob', groupId: 1 });
+     *   User.cache.add({ id: 2, name: 'Sam', groupId: null });
+     *   User.cache.add({ id: 1, name: 'Bobby', groupId: 1 });
+     * });
```

Comment:
What happens if you try to nest batches?

Reply:
They all run in the same batch

Reply:
If not already there, can you double check that test cases confirm this works as expected

Reply:
explicit test case added